### ISSUE TITLE
sql: unknown driver "postgres" (forgotten import?)

### ISSUE THREAD

**Issue by vendion** [bmizerany/pq.go#15]
With using the version of the pq.go driver from the "over" branch I can get the package to build and install, but when I import the driver and try to connect the connection fails with this error:

`panic: sql: unknown driver "postgres" (forgotten import?)`

the way that I am importing it is as follows:

``` go
import (
    "fmt"
    "os"
    "database/sql"
    _ "github.com/bmizerany/pg.go"
)
```

My connect statement is this line:

`db, err := sql.Open("postgres", "dbname=pgtest user=pguser password=password port=5432")`

My version of Go is:

go version weekly.2012-03-04 +5c664f190e4e

Am I doing something wrong or is this an issue with the driver?

**Comment by bmizerany** [comment]
I renamed it to github.com/bmizerany/pq

Please try it on the over branch and let me know if you have the same problem.

**Comment by vendion** [comment]
When did you rename the package because I built it today and the package is still named github.com/bmizerany/pq.go.a

```
ls github.com/bmizerany/
pq.go.a
```

I manually cloned the repository and checkout the over branch and ran `go install` on that.  Here is what I am testing with:

``` go
package main

import (
        "os"
        "database/sql"
        _ "github.com/bmizerany/pq"
)

func main() {
        _, err := sql.Open("postgres", "dbname=pgtest user=pguser password=password port=5432")
        if err != nil {
                panic(err)
        }
        os.Exit(0)
}
```

**Comment by vendion** [comment]
Ah it is because the repository is named pq.go, when `go get` is ran or if a `git clone` is done without explicitly naming the directory name then when the package gets built Go assumes the package name from the directory name and not the `package pq` line in the conn.go file.

so a work around for this would have to be something like:

```
git clone git://github.com/bmizerany/pq.go.git pq
git branch over && git checkout over
git pull origin over
go install .
```

or a very destructive (due to the number of forks your project has) renaming of the repository.

**Comment by bmizerany** [comment]
No. `bmizerany/pq.go` is now one file a major syntax error, on
purpose. Use `bmizerany/pq`.

On Wed, Mar 7, 2012 at 12:38 PM, Adam Jimerson
reply@reply.github.com
wrote:

**Comment by vendion** [comment]
Sorry what do you mean by `bmizerany/pq.go` is a major syntax error on
purpose?

As it stands now if some one was to run `go get
github.com/bmizerany/pq.go` or `git clone git://
github.com/bmizerany/pq.go.git` they get the following directory layout
in their $GOPATH `github.com/bmizerany/pq.go` from there there in the
user runs `go {install|build} github.com/bmizerany/pq.go` they get a
package named pq.go.a.  If the pq.go directory were to be renamed `pq.go
-> pq` and then `go {install|build} github.com/bmizerany/pq` then
they get a package named pq.a.

It seems another way to build or install the package would be to do `go
{install|build} github.com/bmizerany/pq.go/conn.go` because that would
cause it to use the package declaration instead.

In short after building the package as `pq.a` instead of `pq.go.a`
and importing that
I was able to connect to my Postgres database and runs sample select
statement.

On Wed, Mar 7, 2012 at 3:54 PM, Blake Mizerany <
reply@reply.github.com

**Comment by bmizerany** [comment]
Shit. Sorry. I got confused. I renamed a bunch of repos yesterday as a
result of the new behavior in `go get`. I forgot I haven't moved over
pq.go. I'm sorry.

You're correct. You need to rename the directory from `pq.go` ro `pq`
or clone it as `pq` into your GOPATH.

As for the syntax error, this is what pq.go will look like when the
move is complete:
https://github.com/bmizerany/pat.go/blob/master/pat.go

On Wed, Mar 7, 2012 at 1:22 PM, Adam Jimerson
reply@reply.github.com
wrote:

**Comment by vendion** [comment]
No worries we all have days like that, glad that this was cleared up.
